server/src/main/java/org/opensearch/cluster/routing/allocation/decider/FilterAllocationDecider.java

```diff
@@ -127,10 +147,26 @@ public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, Routing
                     "initial allocation of the shrunken index is only allowed on nodes [%s] that hold a copy of every shard in the index";
                 return allocation.decision(Decision.NO, NAME, explanation, initialRecoveryFilters);
             }
+
+            Decision decision = isMixedModeReplicaDecision(shardRouting, allocation);
+            if (decision != null) return decision;
         }
         return shouldFilter(shardRouting, node.node(), allocation);
     }
 
+    public Decision isMixedModeReplicaDecision(ShardRouting shardRouting, RoutingAllocation allocation) {
+        assert shardRouting.unassigned();
+        if (shardRouting.primary() == false
+            && shardRouting.unassignedInfo().getReason() != UnassignedInfo.Reason.INDEX_CREATED
+            && (compatibilityMode.equals(RemoteStoreNodeService.CompatibilityMode.MIXED))
+            && (migrationDirection.equals(RemoteStoreNodeService.Direction.REMOTE_STORE))) {
```

Comment:
We also need to check if primary is migrated or not, right?

Reply:
No, for primary allocation, filter deciders can come in action as is.

Reply:
Added it now .